### PR TITLE
Fix for inexistent `environb` on some platforms

### PR DESCRIPTION
**Pull request by RasmusLar**
Background:
We run prysk on a bash shell in Windows, using a Windows platform Python. Windows Python `os` module does not have environb, so it stopped working after https://github.com/Nicoretti/prysk/pull/177

### CODE REVIEW

**Comment by RasmusLar** [comment]
As Windows isn't tested, and I assume not fully supported, I didn't write any tests for this, and I dunno how it could be tested properly without running the tests on a platform that doesn't support `environb`.. Open for any ideas and I can try to solve it!

File: prysk/test.py [hunk 1]
```diff
@@ -217,7 +217,10 @@ def create_environment(environment, shell, clean=False):
                 out = _escape(out)
 
             try:
-                tmpdir = os.environb[b"TMPDIR"]
+                if os.supports_bytes_environ:
+                    tmpdir = os.environb[b"TMPDIR"]
+                else:
+                    tmpdir = str.encode(os.environ["TMPDIR"])
```
Comment (dlax) [inline prysk/test.py:223]:
```suggestion
                    tmpdir = os.environ["TMPDIR"].encode()
```

**Review by dlax — COMMENTED** [review]
Indeed, while working on #177, I didn't think windows was supported.

The fix here looks good to me.

**Comment by Nicoretti** [comment]
Hi @RasmusLar and @dlax,

thanks for the contributions and support.

Regarding windows, it isn't officially supported.
Also I am not very eager to change that (I don't even have a windows machine).

That said, if the project gets patches improving platform compatibility, I don't see why
we should not apply them.

If anyone with sufficient windows knowledge would contribute a workflow to validate 
windows compatibility that would be fine too.

:information_source: Full disclosure
I am not sure how far I would go in terms of trade-offs to keep windows compatibility.
So passing such a workflow would not be mandatory.


I'll try to merge and release the change the next days.

best
Nico

File: prysk/test.py [hunk 2]
```diff
@@ -217,7 +217,10 @@ def create_environment(environment, shell, clean=False):
                 out = _escape(out)
 
             try:
-                tmpdir = os.environb[b"TMPDIR"]
+                if os.supports_bytes_environ:
+                    tmpdir = os.environb[b"TMPDIR"]
+                else:
+                    tmpdir = os.environ["TMPDIR"].encode()
```
Comment (Nicoretti) [inline prysk/test.py:223]: @dlax @RasmusLar  shouldn't the `else` branch work for all win and unix platforms?

**Comment by dlax** [comment]
>  shouldn't the `else` branch work for all win and unix platforms?


Probably; but keeping the 'if' seems better to me (if that's what you're wondering).

**Comment by Nicoretti** [comment]
> shouldn't the `else` branch work for all win and unix platforms?
> Probably; but keeping the 'if' seems better to me (if that's what you're wondering).

Yes that was the reason.
I think that a single code path results in less code and less reasoning if and when a certain code path is executed.

I mean we can do it either way in this case, still I am curious why keeping the if is better from your point of view?

**Comment by dlax** [comment]
Using environb avoids the encoding step, and any possible error during that operation. But keeping a single code path is also a valid design, sure (and until we get a case for such an error, it seems safe to me).

**Pull request merged**